bindings/python/pyiree/compiler2/tf.py

```diff
@@ -0,0 +1,168 @@
+# Lint-as: python3
+"""TensorFlow compiler interface."""
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from enum import Enum
+import tempfile
+from typing import List, Optional, Sequence, Set, Union
+
+from .tools import *
+from .core import CompilerOptions, DEFAULT_TESTING_BACKENDS, build_compile_command_line
+
+__all__ = [
+    "compile_saved_model",
+    "compile_module",
+    "is_available",
+    "DEFAULT_TESTING_BACKENDS",
+    "ImportOptions",
+    "ImportType",
+]
+
+_TF_IMPORT_TOOL = "iree-tf-import"
+
+
+def is_available():
+  """Determine if TensorFlow and the compiler are available."""
+  try:
+    import tensorflow as tf
+  except ModuleNotFoundError:
+    return False
+  try:
+    find_tool(_TF_IMPORT_TOOL)
+  except ValueError:
+    return False
+  return True
+
+
+class ImportType(Enum):
+  """Import type of the model."""
+  OBJECT_GRAPH = "savedmodel_v2"
+  V2 = "savedmodel_v2"
+  SIGNATURE_DEF = "savedmodel_v1"
+  V1 = "savedmodel_v1"
+
+  @staticmethod
+  def parse(spec) -> "ImportType":
+    if isinstance(spec, ImportType):
+      return spec
+    spec = spec.upper()
+    if spec not in ImportType.__members__:
+      raise ValueError(f"For import_type= argument, expected one of: "
+                       f"{', '.join(ImportType.__members__.keys())}")
+    return ImportType[spec]
+
+
+class ImportOptions(CompilerOptions):
+  """Import options layer on top of the backend compiler options."""
+
+  def __init__(self,
+               exported_names: Sequence[str] = (),
```

Comment:
```suggestion
  def __init__(self,
               *,
               exported_names: Sequence[str] = (),
```

Thought this was a nice touch in `CompilerOptions`.